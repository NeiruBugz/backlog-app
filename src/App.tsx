import 'normalize.css/normalize.css';
import './App.css';
import { useCallback, useState } from 'react';

import { Header, Card, AuthModal } from '@components';
import { useEscapeKeyPress } from './hooks/useEscapeKeyPress';
import { cardProps } from './mocks/data/card.mock';

function App() {
  const [modalOpen, setModalOpen] = useState(false);
  useEscapeKeyPress(useCallback(() => setModalOpen(false), []));

  return (
    <>
      <div className="container">
        <Header setModalOpen={setModalOpen} />
        <Card {...cardProps} />
      </div>
      <AuthModal isOpen={modalOpen} />
    </>
  );
}

export default App;
