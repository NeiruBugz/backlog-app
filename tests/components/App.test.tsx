import { render } from '@testing-library/react';
import App from '../../src/App';

describe('App Component', () => {
  it('snapshot', () => {
    const { container } = render(<App/>);
    expect(container).toMatchSnapshot();
  });
});