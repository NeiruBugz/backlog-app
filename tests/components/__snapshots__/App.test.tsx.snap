// Vitest Snapshot v1

exports[`App Component > snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <header
      class="ba-header"
    >
      <a
        href="src/ui/components#"
      >
        Backlog App
      </a>
      <p
        class="ba-header__user"
      >
        NeiruBugz
      </p>
    </header>
    <div
      class="ba-card"
    >
      <div
        class="ba-card__image"
        style="background-image: url(https://assets1.ignimgs.com/2020/02/14/persona-5-royal---button-fin-1581716582492.jpg);"
      />
      <div
        class="ba-card__info"
      >
        <p
          class="ba-card__title"
        >
          Persona 5 Royal
        </p>
        <span
          class="ba-card__platform ba-card__platform--nintendo"
        >
          Nintendo Switch
        </span>
      </div>
    </div>
  </div>
</div>
`;
